### PR TITLE
Add option to checkout solution repository for Haskell exercises.

### PR DESCRIPTION
**Pull request by sleiss**
### Checklist
- [ ] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
For Haskell exercises, it sometimes is required to check the student submission by comparing it to the sample solution.
This is already possible now, but only after manual setup of Bamboo.

### Description
This PR adds an option to checkout the solution repository (into a `solution` folder) during testing. The test cases can then execute both the students submission and the sample solution and compare the output (e.g. when testing with random input).

**Note**: It's the instructor's responsibility to ensure that the student's code can not leak the sample solution.

### Steps for Testing
1. Create a Haskell programming exercise and enable the checkout of the solution repository (see screenshots below).
2. Open both the base and solution build plans in Bamboo, and verify that the solution repository is added:
Peek 2020-10-16 13-46
3. Check the tasks and verify that the solution repository is checked out into the `solution` path.
Peek 2020-10-16 13-47
4. Create one Haskell programming exercise and one Java programming exercise without enabling this option and verify that the solution repository is **not** added to the repositories (see step 2) and checkout task (see step 3).

### Test Coverage

* ExerciseService.java: 85%
* programming-exercise.component.ts 95%

### Screenshots
Added an option during exercise creation that is only available for Haskell exercises.
image
**Note**: As we do not persist whether the solution repository should be checked out, we do not show the value of this flag in the `View` and `Edit` dialogs.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java:179]: does this mean the solution repo is checked out twice in the SOLUTION build plan?

Comment (sleiss) [reply src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java:179]: @krusche Yes: Once as `assignment` and once as `solution`. I believe this is useful as otherwise the solution would require special tests.

Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java:179]: ok

File: src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseIntegrationTest.java
Comment (krusche) [inline src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseIntegrationTest.java:747]: can we improve this test using the programming language as enum parameter so that all programming languages except Haskell are tested to produce a bad request? (this would then also include future programming languages

File: src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java:104]: Can you please document the property and its default value here? thanks!

File: src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java:35]: please add a hint that this is only used for special cases

**Comment by krusche** [comment]
The general structure of the PR looks good. It would be nice if some people from the programming exercise team (@kloessst @schultek @fde312 @MaisiKoleni) could test this PR. As soon as we have 3 approvals, we can merge it. Then, the instructors of the actual courses could further test the implementation on the production server.

File: src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseIntegrationTest.java
Comment (sleiss) [reply src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseIntegrationTest.java:747]: I've changed it to a `@ParameterizedTest` that uses all `ProgrammingLanguage`s except `HASKELL`. This is also safe for the future as the test excludes `HASKELL` (as opposed to including all other programming languages).

File: src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java
Comment (sleiss) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java]: I've added documentation.

File: src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java
Comment (sleiss) [reply src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java:110]: I've added documentation.

**Review by krusche — APPROVED** [review]

**Pull request merged**
